### PR TITLE
HPCC-14330 ConfigMgr - Add cassandra config options

### PR DESCRIPTION
**Pull request by garonsky**
Signed-off-by: Gleb Aronsky gleb.aronsky@lexisnexis.com

### CODE REVIEW

**Comment by garonsky** [comment]
@rpastrana please review

**Comment by richardkchapman** [comment]
Did you end up changinf the format of the info stored in the environment in the end?

File: initfiles/componentfiles/configxml/dali.xsd [hunk 1]
```diff
@@ -31,6 +31,39 @@
...
+          <xs:complexType>
+          <xs:sequence>
+            <xs:element name="Options"/>
+          </xs:sequence>
+            <xs:attribute name="name" type="xs:string" use="required" default="name">
+              <xs:annotation>
+                <xs:appinfo>
+                  <colIndex>1</colIndex>
+                  <autogenforwizard>1</autogenforwizard>
+                  <autogendefaultvalue>server</autogendefaultvalue>
```
Comment (rpastrana) [inline initfiles/componentfiles/configxml/dali.xsd]: Will it make sense to default server|local host for every new instance?

File: initfiles/componentfiles/configxml/dali.xsd [hunk 2]
```diff
@@ -31,6 +31,39 @@
...
       <xs:sequence>
+
+        <xs:element name="PluginOptions" maxOccurs="unbounded">
+          <xs:annotation>
+            <xs:appinfo>
+              <title>Plugin Options</title>
+            </xs:appinfo>
+          </xs:annotation>
+          <xs:complexType>
+          <xs:sequence>
```
Comment (rpastrana) [inline initfiles/componentfiles/configxml/dali.xsd]: trivial, but it looks like this section should be indented

File: initfiles/componentfiles/configxml/dali.xsd [hunk 3]
```diff
@@ -31,6 +31,39 @@
...
+                 <colIndex>2</colIndex>
+                 <autogenforwizard>1</autogenforwizard>
+                 <autogendefaultvalue>localhost</autogendefaultvalue>
+               </xs:appinfo>
+              </xs:annotation>
+            </xs:attribute>
+          </xs:complexType>
+        </xs:element>
+
+
```
Comment (rpastrana) [inline initfiles/componentfiles/configxml/dali.xsd]: Trivial, but there might be extra lines here.

**Comment by garonsky** [comment]
@rpastrana Changes made based on comments.
@richardkchapman as per myemail, this commit will modify the environment.xml to support this format (sample config):

```
  <DaliServerProcess build="_"
                     buildSet="dali"
                     pluginentrypoint="createWorkUnitFactory"
                     environment="/etc/HPCCSystems/environment.xml"
                     name="mydali"
                     pluginname="cassandraembed"
                     recoverFromIncErrors="true"
                     plugintype="WorkunitServer">
   <Instance computer="node001001"
             directory="/var/lib/HPCCSystems/mydali"
             name="s1"
             netAddress="1.1.1.1"
             port="7070"/>
   <PluginOptions name="server" value="localhost"/>
   <PluginOptions name="randomWuidSuffix" value="4"/>
   <PluginOptions name="tracelevel" value="1"/>
   <PluginOptions name="keyspace" value="hpcc"/>
  </DaliServerProcess>
```

**Comment by richardkchapman** [comment]
Sorry, I didn't see that email. But that format is not good, as it assumes there can only be one plugin.

Better would be to have a separate DaliServerPlugin section, with a reference back to the dali it is acting as a plugin for.

i.e.

<DaliServerProcess build="_"
                     buildSet="dali"
                     environment="/etc/HPCCSystems/environment.xml"
                     name="mydali"
                     recoverFromIncErrors="true">
   <Instance computer="node001001"
             directory="/var/lib/HPCCSystems/mydali"
             name="s1"
             netAddress="1.1.1.1"
             port="7070"/>
  </DaliServerProcess>
  </DaliServerPlugin name="whatever" entrypoint="createWorkUnitFactory" dali="mydali" pluginName="cassandraembed" type="workunitServer">
   <Optionname="server" value="localhost"/>
   <Option name="randomWuidSuffix" value="4"/>
   <Option name="tracelevel" value="1"/>
   <Option name="keyspace" value="hpcc"/>
  </DaliServerProcess>

**Comment by richardkchapman** [comment]
@garonsky Back to you

**Comment by garonsky** [comment]
Will re-open when updated.

**Pull request closed without merging**